src/Language/PureScript/Docs/Convert.hs

```diff
@@ -97,17 +97,51 @@ insertValueTypesAndInferredKinds env m =
       key = P.Qualified (Just (modName m)) (P.ProperName name)
     in case Map.lookup key (P.types env) of
       Just (inferredKind, _) ->
-        d { declKind = Just $ KindInfo { kiKeyword = keyword, kiKind = inferredKind' } }
+        if isFunctionTypeStarTerminal inferredKind'
+          then  d
+          else  d { declKind = Just $ KindInfo
+                    { kiKeyword = keyword
+                    , kiKind = dropTypeSortAnnotation inferredKind'
+                    }
+                  }
         where
-          inferredKind' = dropTypeSortAnnotation $ inferredKind $> ()
+          inferredKind' = inferredKind $> ()
           kindPrimType = P.TypeConstructor () Prim.Type
+          kindPrimFunction = P.TypeConstructor () Prim.Function
+          kindPrimConstraint = P.TypeConstructor () Prim.Constraint
+
+          -- `Type ->`
+          kindFunctionType = P.TypeApp () kindPrimFunction kindPrimType
+
+          -- |
+          -- Returns True if the inferred kind signature follows this
```

Comment:
How about calling this function `isUninteresting` and saying that an "uninteresting" kind is a kind of the form:
- `Type`
- `Constraint`
- `Type -> K` where `K` is uninteresting.
I think that makes it clearer which kinds are included, and I think it will make the code a little easier to read.

Reply:
Ah, thanks! That's a much better name.